### PR TITLE
docs(config): add `experimentalScopedSlotChanges` extras flag

### PR DESCRIPTION
**Pull request by tanner-reits**
Adds a new `experimentalScopedSlotChanges` extras config option to toggle the new behavior for `scoped` components.

Related code changes: https://github.com/ionic-team/stencil/pull/5220

**Note:** The first commit only alphabetizes the options, but results in what looks like many more changes

### CODE REVIEW

File: docs/config/extras.md
Comment (tanner-reits) [inline docs/config/extras.md:73]: I'll make sure these versions are accurate before this merges, just depends how this lines up with upcoming releases

**Review by rwaskiewicz — COMMENTED** [review]
Overall LGTM - I'll give it another pass once I get to/through the impl PRs

**Review by christian-bromann — APPROVED** [review]
👍

**Review by rwaskiewicz — APPROVED** [review]
I approved the impl PR, I'm good with approving now and just landing it after you've bumped the version in the docs

**Pull request merged**
